### ISSUE TITLE
conflicting precision issues

### ISSUE THREAD

**Issue by alexpineda** [vanruesc/postprocessing#274]
### Description of the bug

I noticed that the depth buffer is using mediump precision due to GL_FRAGMENT_PRECISION_HIGH not being set. I'm confused that it just pertains to the depth buffer and not other items that work with highp. I'm specifically getting an error of
```
ERROR: 0:346: 'e1ComputeWorldPosition' : no matching overloaded function found
ERROR: 0:346: '=' : dimension mismatch
ERROR: 0:346: '=' : cannot convert from 'const mediump float' to 'highp 3-component vector of float'
ERROR: 0:350: 'e1GetFog' : no matching overloaded function found
```
I'm not quite sure what to do about it. I do not think it's due to a bug but simply me not understanding how to integrate the depth buffer information into the rest of my program.

### To Reproduce

The resulting glsl program output, with the problematic line highlighted.

https://gist.github.com/alexpineda/ca5d230ffcb0e358e42f443ef0848f3f#file-fog-glsl-L345

### Library versions used

 - Three: 0.125.2
 - Post Processing: 6.19.3

### Desktop

 - OS: Windows
 - Browser: Chrome/Electron
 - Graphics hardware: RTX2080

**Comment by vanruesc** [comment]
`GL_FRAGMENT_PRECISION_HIGH` is a built-in WebGL constant that won't appear in the shader code. If this macro is not defined, all floats and vectors will be `mediump` or `lowp` in the fragment shader.

The error indicates that the shader is trying to assign a `float` value to a `vec3` variable. Line 346, however, is empty, so this error doesn't match up with the code. Please consider providing a reproduction of the issue via [CodeSandbox](https://codesandbox.io/s/postprocessing-25rts) or some other service.

**Comment by alexpineda** [comment]
Thank you. I've forked a sandbox here. https://codesandbox.io/s/adoring-violet-m6635?file=/src/App.js Unfortunately the environment is causing an issue with the ES6 class I added :S.

**Comment by vanruesc** [comment]
Thanks for preparing that example!

The compiler was complaining because GLSL doesn't support function hoisting like JavaScript; it's more like C where functions need to be declared before they are used in code. There was also a duplicate `color` uniform which had to be removed.

This fixed the compiler errors, but the effect still renders a black image because there still seem to be some bugs:
https://codesandbox.io/s/recursing-mcclintock-rz7si?file=/src/FogOfWarEffect.js

I've also fixed the ES6 issue which was caused by a missing babel transpilation step - thanks for pointing this out earlier!
